sql_extensions/migrations/special-queries.sql

```diff
@@ -47,15 +47,54 @@ WHERE
 ORDER BY
     (data -> 'market_metadata' ->> 'market_id')::NUMERIC,
     (data ->> 'transaction_version')::NUMERIC DESC,
-    (data ->> 'event_index')::NUMERIC DESC;
+    (data -> 'state_metadata' ->> 'market_nonce')::NUMERIC DESC;
 
 CREATE OR REPLACE FUNCTION UPDATE_LATEST_STATE()
 RETURNS TRIGGER AS $$
 BEGIN
-  DELETE FROM inbox_latest_state
-  WHERE (inbox_latest_state.data->'market_metadata'->>'market_id')::numeric = (NEW.data->'market_metadata'->>'market_id')::numeric
-  AND (inbox_latest_state.data->'state_metadata'->>'bump_time')::numeric <= (NEW.data->'state_metadata'->>'bump_time')::numeric;
-  INSERT INTO inbox_latest_state SELECT NEW.*;
+  IF EXISTS (
+    SELECT 1
+    FROM inbox_latest_state
+    WHERE (inbox_latest_state.data->'market_metadata'->>'market_id')::numeric = (NEW.data->'market_metadata'->>'market_id')::numeric
+      AND (inbox_latest_state.data->'state_metadata'->>'market_nonce')::numeric < (NEW.data->'state_metadata'->>'market_nonce')::numeric
+  ) THEN
+    RETURN NEW;
+  END IF;
+
```

Comment:
1. check uniqueness by `market_nonce`
2. run the check before trying to insert / on conflict, to avoid deadlocks (I was running into them until I changed it to this)
3. aka break early to avoid unnecessary resource contention

Reply:
I am curious about these deadlocks. Can you post any error messages you were getting ?